test_defopt.py

```diff
@@ -307,6 +307,14 @@ def test_bool_kwarg(self):
         def main(foo='default'):
             """:type foo: bool"""
             return foo
+        self.assertIs(defopt.run(main, options='has_default',
+                                 argv=[]), 'default')
+        self.assertIs(defopt.run(main, options='has_default',
+                                 argv=['--foo']), True)
+        self.assertIs(defopt.run(main, options='has_default',
+                                 argv=['--no-foo']), False)
+        self.assertIs(defopt.run(main, options='has_default',
+                                 argv=['--foo', '--no-foo']), False)
         self.assertIs(defopt.run(main, strict_kwonly=False,
```

Comment:
I would additionally wrap all of these with self.assertWarns(DeprecationWarning): ...